src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Mappers/Apprentice/ConfirmEditApprenticehsipRequestToConfirmEditViewModelMapperTests.cs

```diff
@@ -0,0 +1,249 @@
+            source.Cost = 1000;
+            source.EmployerReference = _apprenticeshipResponse.EmployerReference;
+            source.StartDate = new CommitmentsV2.Shared.Models.MonthYearModel(_apprenticeshipResponse.StartDate.Month.ToString() + _apprenticeshipResponse.StartDate.Year) ;
+            source.EndDate = new CommitmentsV2.Shared.Models.MonthYearModel(_apprenticeshipResponse.EndDate.Month.ToString() + _apprenticeshipResponse.EndDate.Year);
+
+            _priceEpisodeResponse = autoFixture.Build<GetPriceEpisodesResponse>()
+                .With(x => x.PriceEpisodes, new List<PriceEpisode> {
+                    new PriceEpisode { Cost = 1000, FromDate = DateTime.Now.AddMonths(-1), ToDate = null}})
+                .Create();
+
+            _standardSummary = autoFixture.Create<TrainingProgramme>();
+            _standardSummary.EffectiveFrom = new DateTime(2018, 1, 1);
+            _standardSummary.EffectiveTo = new DateTime(2022, 1, 1);
+            _standardSummary.FundingPeriods = SetPriceBand(1000);
+
+            _mockCommitmentsApiClient = new Mock<ICommitmentsApiClient>();
+
+            _mockCommitmentsApiClient.Setup(c => c.GetApprenticeship(It.IsAny<long>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(_apprenticeshipResponse);
+            _mockCommitmentsApiClient.Setup(c => c.GetPriceEpisodes(It.IsAny<long>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(_priceEpisodeResponse);
+            _mockCommitmentsApiClient.Setup(t => t.GetTrainingProgramme(_apprenticeshipResponse.CourseCode, It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new GetTrainingProgrammeResponse
+                {
+                    TrainingProgramme = _standardSummary
+                });
+
+            _encodingService = new Mock<IEncodingService>();
+            _encodingService.Setup(x => x.Decode(It.IsAny<string>(), It.IsAny<EncodingType>())).Returns(22);
+            _mapper = new ConfirmEditApprenticehsipRequestToConfirmEditViewModelMapper(_mockCommitmentsApiClient.Object, _encodingService.Object);
+        }
+
+        public List<TrainingProgrammeFundingPeriod> SetPriceBand(int fundingCap)
+        {
+            return new List<TrainingProgrammeFundingPeriod>
+            {
+                new TrainingProgrammeFundingPeriod
+                {
+                        EffectiveFrom = new DateTime(2019, 1, 1),
+                        EffectiveTo = DateTime.Now.AddMonths(1),
+                        FundingCap = fundingCap
+                }
+            };
+        }
+
+        public async Task<ConfirmEditApprenticeshipViewModel> Map()
+        {
+          resultViewModl =  await _mapper.Map(source);
+            return resultViewModl;
+        }
+
+        internal void VerifyCommitmentApiIsCalled()
+        {
+            _mockCommitmentsApiClient.Verify(c => c.GetApprenticeship(It.IsAny<long>(), It.IsAny<CancellationToken>()), Times.Once());
```

Comment:
could verify correct apprenticeshipId is passed to the mocked call

Reply:
done